.github/workflows/ci.yml

```diff
@@ -93,33 +93,9 @@ jobs:
                   fail_level: any
                   filter_mode: diff_context
 
-    check-api:
-        runs-on: ubuntu-latest
-        permissions:
-            contents: read
-            checks: write
-            issues: write
-            pull-requests: write
-        steps:
-            - uses: actions/checkout@v5
-              with:
-                  fetch-depth: 0
-                  fetch-tags: true
-            - uses: actions/setup-java@v5
-              with:
-                  distribution: temurin
-                  java-version: 21
-            - run: ./gradlew updateLegacyAbi
-            - if: ${{ github.event_name == 'pull_request' }}
-              uses: reviewdog/action-suggester@v1
-              with:
-                  tool_name: api-dump
-                  fail_level: any
-                  filter_mode: added
-
```

Comment:
extra: integrated this into pre-commit so we don't need the separate task anymore